### PR TITLE
Flux standard action rule

### PR DESCRIPTION
**Pull request by ahuth**
RFC -> https://github.com/mavenlink/rfc/pull/34

This implements
- the `use-flux-standard-actions` rule, enforcing Flux Standard Actions.
- packages and setup needed to test the lint rule

Note that right now this only checks object literal actions that have a `type` property, and will not consider thunk actions. This is to reduce false positives as I haven't figured out how to determine in a lint rule whether or not something is actually exported or not.

### CODE REVIEW

**Comment by ahuth** [comment]
@brandonduff @jsteckml @naganowl @juanca, can you take a look at this please?

Most of the diff is yarn.lock changes.

There's also 
- a new ESLint rule
- setup needed to test the rule with Jest

File: packages/eslint-plugin-mavenlint/rules/use-flux-standard-actions.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        const extraKeys = keys.filter(key => !standardKeys.includes(key));
+        if (extraKeys.length > 0) {
+          context.report({
+            node,
+            message: 'Unexpected keys: {{ unexpectedKeys }}. Generally, redux actions should only have type, payload, error, and meta properties',
+            data: {
+              unexpectedKeys: extraKeys.join(', '),
+            },
+          });
+        }
```
Comment (jsteckml) [inline packages/eslint-plugin-mavenlint/rules/use-flux-standard-actions.js]: Would it be useful to try to optimize lint rules, in general?

Comment (ahuth) [reply packages/eslint-plugin-mavenlint/rules/use-flux-standard-actions.js]: Definitely seems like that would be useful. We can get timing data from ESLint:

```
Rule                                | Time (ms) | Relative
:-----------------------------------|----------:|--------:
import/no-named-as-default          |  3907.034 |     9.0%
indent                              |  3869.342 |     8.9%
react/void-dom-elements-no-children |  3585.559 |     8.3%
import/no-duplicates                |  3109.513 |     7.2%
import/no-named-as-default-member   |  2046.964 |     4.7%
react/prop-types                    |  1174.247 |     2.7%
react/no-unused-prop-types          |  1158.527 |     2.7%
react/no-deprecated                 |   982.487 |     2.3%
react/prefer-stateless-function     |   982.456 |     2.3%
react/no-string-refs                |   948.852 |     2.2%
```

And this new rule doesn't show up in there. Definitely makes sense to keep track and make linting as fast as possible, @jsteckml

Comment (ahuth) [reply packages/eslint-plugin-mavenlint/rules/use-flux-standard-actions.js]: Was also able to measure the performance of just the new rule with:

```
TIMING=true yarn lint --no-eslintrc --plugin "mavenlint" --rule "mavenlint/use-flux-standard-actions: 2" --parser "babel-eslint"
```

which generated:

```
Rule                                | Time (ms) | Relative
:-----------------------------------|----------:|--------:
mavenlint/use-flux-standard-actions |    18.027 |   100.0%
```

Not sure if this is telling me the average time spent in the rule or what

**Comment by ahuth** [comment]
Mergin' this thing. Let me know if anyone has any issues with anything here, and I'm happy to discuss more and adjust as necessary.

**Pull request merged**